File: wallet_syncronizer/pkg/service/token/get/get.go
package get

import (
	"errors"
	"github.com/gofiber/fiber/v2"
	"gorm.io/gorm"
	token_db "wallet-syncronizer/pkg/database/token"
)

type Service struct {
	db      *gorm.DB
	tokenId string
}

func NewService(db *gorm.DB, tokenId string) *Service {
	return &Service{
		db:      db,
		tokenId: tokenId,
	}
}

func (s Service) Get() (status int, token *token_db.Token, err error) {
	token = new(token_db.Token)
	if err = s.db.Where("TokenId = ?", s.tokenId).First(token).Error; err != nil {
		if errors.Is(err, gorm.ErrRecordNotFound) {
			return fiber.StatusNotFound, nil, err
		} else {
			return fiber.StatusInternalServerError, nil, err
		}
	}
	return fiber.StatusOK, token, nil
}

File: wallet_syncronizer_node/service/token/get/get.js
const { Token } = require('../../../database/token');

class GetService {
    constructor(sequelize, tokenId) {
        this.sequelize = sequelize;
        this.tokenId = tokenId;
    }

    async get() {
        const token = await Token.findOne({
            where: {
                tokenId: this.tokenId,
            },
        });

        if (!token) {
            return {
                status: 404,
                token: null,
                err: new Error('Token not found'),
            };
        }

        return {
            status: 200,
            token,
            err: null,
        };
    }
}

module.exports = GetService;